plugins/jenkins-backend/README.md

```diff
@@ -0,0 +1,160 @@
+# Jenkins Plugin (Alpha)
+
+Welcome to the jenkins backend plugin! Website: [https://jenkins.io/](https://jenkins.io/)
+
+This is the backend half of the 2 jenkins plugins and is responsible for:
+
+- finding an appropriate instance of jenkins for an entity
+- finding the appropriate job(s) on that instance for an entity
+- connecting to jenkins and gathering data to present to the frontend
+
+## Integrating into a backstage instance
+
+This plugin needs to be added to an existing backstage instance.
+
+Typically, this means creating a `src/plugins/jenkins.ts` file and adding a reference to it to `src/index.ts`
+
+### jenkins.ts
+
+```typescript
+import {
+  createRouter,
+  SingleJenkinsInfoProvider,
+} from '@backstage/plugin-jenkins-backend';
+import { Router } from 'express';
+import { PluginEnvironment } from '../types';
+
+export default async function createPlugin({
+  logger,
+}: PluginEnvironment): Promise<Router> {
+  return await createRouter({
+    logger,
+    jenkinsInfoProvider: new SingleJenkinsInfoProvider(),
+  });
+}
+```
+
+This plugin must be provided with a JenkinsInfoProvider, this is a strategy object for finding the jenkins instance and job for an entity.
+
+There is a selection of standard ones provided, but the Integrator is free to build their own.
+
+### SingleJenkinsInfoProvider
+
+Allows configuration of a single global jenkins instance and annotating entities with the job name on that instance.
+
+The following will look for jobs for this entity at `https://jenkins.example.com/job/teamA/job/artistLookup-build`
+
+#### Config
+
+```yaml
+jenkins:
+  baseUrl: https://jenkins.example.com
+  username: backstage-bot
+  apikey: 123456789abcdef0123456789abcedf012
+```
+
+#### Catalog
+
+```yaml
+apiVersion: backstage.io/v1alpha1
+kind: Component
+metadata:
+  name: artist-lookup
+  annotations:
+    'jenkins.io/github-folder': teamA/artistLookup-build
```

Comment:
I went with job-slug as while this will most often be a "folder", it will sometimes be a "project". Both "folder"s and "project"s are "job"s in Jenkins parlance.